### PR TITLE
Init commit for swbd

### PR DESCRIPTION
**Pull request by JinZr**
This pull request creates the init commit for the Switchboard dataset. It is currently very rough and produces a high Word Error Rate, requiring more improvement and refinement. 

More info please refer to the README and RESULTS markdown files.

### CODE REVIEW

File: egs/swbd/ASR/RESULTS.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+The best WER, as of 2023-09-04, for the Switchboard is below
+
+Results using attention decoder is given as:
+
+|                                |  eval2000-swbd  |  eval2000-callhome  | eval2000-avg |
+|--------------------------------|-----------------|---------------------|--------------|
+|         `conformer_ctc`        |      9.48       |         17.73       |    13.67     | 
+
+Decoding results and models can be found here:
+https://huggingface.co/zrjin/icefall-asr-swbd-conformer-ctc-2023-8-26
```
Comment (csukuangfj) [inline egs/swbd/ASR/RESULTS.md:15]: Please also upload the following files:
- the last checkpoint, e.g., epoch-100.pt
- the script for exporting the models
- a folder `test_wavs` containing some test waves for this model
- a README.md containing a link to this pull-request in icefall

File: egs/swbd/ASR/conformer_ctc/conformer.py [hunk 2]
```diff
@@ -0,0 +1,910 @@
+#!/usr/bin/env python3
```
Comment (csukuangfj) [inline egs/swbd/ASR/conformer_ctc/conformer.py:1]: Please replace it with a symlink.

File: egs/swbd/ASR/RESULTS.md (hunk 1 above)
Comment (csukuangfj) [inline egs/swbd/ASR/RESULTS.md:15]: Please also add a CI test for it.

**Comment by JinZr** [comment]
Dear Desh,

This PR is pending review now and should be able to merge at any moment. I
intend to make a separate PR for Zipformer as there are still some
parameters to tune.

Best Regards


Desh Raj ***@***.***>于2023年9月28日 周四22:28写道：

**Review by csukuangfj — APPROVED** [review]
Thanks!

**Comment by JinZr** [comment]
@desh2608 Dear Desh,

The PR has been merged, please check.

Thank you!

Best

**Comment by huangruizhe** [comment]
Hello, I was wondering about the sampling rate of swbd data in the recipe:
https://github.com/k2-fsa/icefall/blob/master/egs/swbd/ASR/local/compute_fbank_swbd.py#L108

The original swbd data is [8kHz](https://github.com/lhotse-speech/lhotse/blob/master/lhotse/recipes/switchboard.py#L4), but [here](https://github.com/k2-fsa/icefall/blob/master/egs/swbd/ASR/local/compute_fbank_swbd.py#L108
) you are specifying 16kHz instead. Is there any reason we do this? Thanks.

**Comment by JinZr** [comment]
Dear Huangruizhe,

the resampling here is for the musan related data augmentation we applied
later, you can skip the resampling stage if you want to get rid of the
musan part.

hope this would help.


Best Regards
Jin


huangruizhe ***@***.***>于2023年10月13日 周五00:42写道：

**Comment by huangruizhe** [comment]
Thanks for the explanation. Is it more natural to downsample MUSAN to 8kHz instead? E.g., https://github.com/pyannote/pyannote-audio/issues/300

If we are using upsampled 16kHz swbd audio to compute features -- fbanks here -- we are probably wasting the 8kHz~16kHz part in the feature, as it contains no information except noise. By speculation, will it degrade the performance?

**Comment by JinZr** [comment]
Dear Huangruizhe,


I haven’t done any experiment in terms of the performance but I suppose
that makes sense, especially when you want a system for narrowband
scenarios.

i’ll work on that later.


Best Regards
Jin


huangruizhe ***@***.***>于2023年10月13日 周五01:39写道：

**Comment by huangruizhe** [comment]
Yes, I agree. Looking forward to the updates.

**Comment by JinZr** [comment]
Dear huangruizhe,

Sorry for kept you waiting, a new narrowband setup is created
https://github.com/k2-fsa/icefall/pull/1391, PR is currently pending
results from the latest experiments.

Stay tuned!

best
jin

**Pull request merged**